test/Npgsql.Tests/CopyTests.cs

```diff
@@ -696,6 +696,40 @@ public void ImportExceptionDoesNotCommit()
             }
         }
 
+        [Test, IssueLink("https://github.com/npgsql/npgsql/issues/2347")]
+        public void Write_ColumnOutOfBounds_ThrowsInvalidOperationException()
+        {
+            using (var conn = OpenConnection())
+            {
+                conn.ExecuteNonQuery("CREATE TEMP TABLE data (field_text TEXT, field_int2 SMALLINT)");
```

Comment:
Changing `SMALLINT` to `INTEGER` will allow you to avoid casts below.

Reply:
Good catch. Copy and pasted from some other test without thinking about it. Changed :)